stub/src/main/java/io/grpc/kotlin/ServerCalls.kt

```diff
@@ -275,17 +276,17 @@ object ServerCalls {
 
       override fun onMessage(message: RequestT) {
         if (isReceiving) {
-          try {
-            if (!requestsChannel.offer(message)) {
+          val result = requestsChannel.trySend(message)
+          isReceiving = result.isSuccess
+          result.onFailure { ex ->
+            if (ex !is CancellationException) {
               throw Status.INTERNAL
                 .withDescription(
                   "onMessage should never be called when requestsChannel is unready"
                 )
+                .withCause(ex)
                 .asException()
             }
```

Comment:
I think this will start wrapping exceptions that were previously thrown directly by `.offer` in `StatusException`.

Code to maintain previous semantics would be

```kotlin
            if (ex !is CancellationException) {
              throw ex ?: Status.INTERNAL
                .withDescription(
                  "onMessage should never be called when requestsChannel is unready"
                )
                .asException()
            }
```